app/controllers/foreclosures_controller.rb

```diff
@@ -1,131 +1,133 @@
-      respond_to do |format|
-        format.html
-        format.csv { send_data @foreclosures.to_csv }
-        format.xls { send_data @foreclosures.to_csv(col_sep: "\t") }
-      end
+    @foreclosures = Foreclosure.all
+
+    respond_to do |format|
+      format.html
+      format.csv { send_data @foreclosures.to_csv } #untested
+      format.xls { send_data @foreclosures.to_csv(col_sep: "\t") } #untested
     end
   end
 
- def new
-  @foreclosure = Foreclosure.new
-end
+  def new
+    @foreclosure = Foreclosure.new
+  end
 
-def search
-  @client = Client.where("first_name LIKE ? OR last_name LIKE ? OR address LIKE ?", "%#{params[:search]}%", "%#{params[:search]}%", "%#{params[:search]}%")
-  render :index
-end
+  def search
+    @client = Client.where("first_name LIKE ? OR last_name LIKE ? OR address LIKE ?", "%#{params[:search]}%", "%#{params[:search]}%", "%#{params[:search]}%")
+    render :index
+  end
 
 
-def create
-  if client_signed_in?
-   @id = current_client.id
- elsif user_signed_in?
-   @id = Foreclosure.where(client_id: params[:id]).first.id
- end
- @foreclosure = Foreclosure.new({ 
-   client_id: @id,
-   currently_foreclosed: params[:currently_foreclosed],
-   originating_lender: params[:originating_lender],
-   original_loan_number: params[:original_loan_number],
-   servicer: params[:servicer], 
-   servicer_loan_number: params[:servicer_loan_number],
-   monthly_mortgage_payment: params[:monthly_mortgage_payment],
-   loan_term: params[:loan_term], 
-   origination_date: params[:origination_date], 
-   been_to_court: params[:been_to_court], 
-   court_case_number: params[:court_case_number], 
-   working_with_lawyer: params[:working_with_lawyer],
-   working_w_agency: params[:working_w_agency], 
-   agency: params[:agency]
-   })
- if @foreclosure.save
-   ProgramEmployee.create({programable_id: @foreclosure.id, programable_type: "Foreclosure"})
-   flash[:success] = ["You've completed the foreclosure application"]
-   redirect_to "/foreclosures/#{@foreclosure.id}"
- else
-  flash[:danger] = @foreclosure.errors.full_messages
-  render :new
-end
-end
+  def create
+
+    @id = current_client.id if current_client
+    @id = Client.find(params[:id]).id if current_user
+
+    @foreclosure = Foreclosure.new({
+      client_id: @id,
+      currently_foreclosed: params[:currently_foreclosed],
+      originating_lender: params[:originating_lender],
+      original_loan_number: params[:original_loan_number],
+      servicer: params[:servicer],
+      servicer_loan_number: params[:servicer_loan_number],
+      monthly_mortgage_payment: params[:monthly_mortgage_payment],
+      loan_term: params[:loan_term],
+      origination_date: params[:origination_date],
+      been_to_court: params[:been_to_court],
+      court_case_number: params[:court_case_number],
+      working_with_lawyer: params[:working_with_lawyer],
+      working_w_agency: params[:working_w_agency],
+      agency: params[:agency]
+      })
```

Comment:
For the update method below...you used the method `foreclosure_params` instead of listing all the attributes. Can we do the same here? The difference is `client_id`.  So you could do something like `Foreclosure.new(foreclosure_params.merge(client_id: @id))`.